common/dialer/resolver.go

```diff
@@ -10,9 +10,10 @@ type Resolver interface {
 	// Resolve resolves a service, getting an address or URL
 	// If a Resolve() call completes successfully but finds no addresses, it will return ("", nil)
 	Resolve() (string, error)
-	// ResolveAll resolves a slice of all known addresses or URLs
-	// If a ResolveAll() call completes successfully but finds no addresses, it will return ([]string{}, nil)
-	ResolveAll() ([]string, error)
+	// ResolveMany resolves a slice of random addresses or URLs
+	// The int parameter specifies the maximum number of addresses to return. If 0, all addresses are returned.
```

Comment:
Given zero values, this interface makes sense, but it still seems awkward... -1 for "return all" might make more sense.

Reply:
Not blocking on this and don't really have a strong opinion, just thinking out loud

Reply:
fair enough - will make it <= 0 to effectively return all (since there's no valid reason to call with 0 anyway)